src/sparseml/transformers/sparsification/sparse_model.py

```diff
@@ -123,21 +128,26 @@ def skip(*args, **kwargs):
 
         # If model is compressed on disk, decompress and load the weights
         if compressor is not None:
-            # if we loaded from a HF stub, find the cached model
-            model_path = get_safetensors_folder(
-                pretrained_model_name_or_path, cache_dir=kwargs.get("cache_dir", None)
+            # decompress weights
+            compressor.overwrite_weights(
+                model_path=pretrained_model_name_or_path, model=model
             )
 
-            # decompress weights
-            compressor.overwrite_weights(model_path=model_path, model=model)
-
-        recipe = resolve_recipe(recipe=recipe, model_path=pretrained_model_name_or_path)
-        if recipe:
-            apply_recipe_structure_to_model(
-                model=model,
-                model_path=pretrained_model_name_or_path,
-                recipe_path=recipe,
+        if quantization_config is not None:
+            # if we loaded from a HF stub, find the cached model
+            apply_quantization_config(model, quantization_config)
+            load_pretrained_quantization(model, pretrained_model_name_or_path)
+        else:
+            recipe = resolve_recipe(
```

Comment:
let's maybe add a short logging message that tells the user that this pathway is going to get obsolete and they should opt in to use the new pathway